File: support_tools/swigwin-3.0.12/Examples/test-suite/octave/director_enum_runme.m
director_enum

MyFoo=@() subclass(director_enum.Foo(),'say_hi',@(self,val) val);

b = director_enum.Foo();
a = MyFoo();

if (a.say_hi(director_enum.hello) != b.say_hello(director_enum.hi))
  error
endif

File: support_tools/swigwin-3.0.12/Examples/test-suite/perl5/director_enum_runme.pl
use strict;
use warnings;
use Test::More tests => 5;
BEGIN { use_ok 'director_enum' }
require_ok 'director_enum';

{
  package MyFoo;
  use base 'director_enum::Foo';
  sub say_hi { my($self, $val) = @_;
    return $val;
  }
}

my $b = director_enum::Foo->new();
isa_ok $b, 'director_enum::Foo';
my $a = MyFoo->new();
isa_ok $a, 'MyFoo';

is $a->say_hi($director_enum::hello),
   $a->say_hello($director_enum::hi);
